source/_integrations/tuya.markdown

```diff
@@ -25,79 +24,123 @@ ha_platforms:
+|1100|param is empty| Empty parameter of username or app. Fill the parameters refer to [Tuya credential table](https://github.com/tuya/tuya-home-assistant/wiki/Install-Tuya-v2?_source=d10de34623e3daca5b02e3c31528a0c4#3-enter-your-tuya-credential).
+|2017|schema does not exist| Incorrect app selected. Select the app you used to link devices in the cloud project.|
+| 2406 | skill id invalid | Make sure that your cloud project on the [Tuya IoT Platform](https://iot.tuya.com/cloud/?_source=fd9f339c3f58e01cdc0b275d5bd97a21) should be created after May 25, 2021. Otherwise, you need to create a new project or migrate data to a new project. For more information, see [Operation on the Tuya IoT Platform](https://developer.tuya.com/en/docs/iot/migrate-from-an-older-version?id=Kamee9wtbd00b#title-3-Operation%20on%20the%20Tuya%20IoT%20Platform).|
+| 28841105 |No permissions. This project is not authorized to call this API| Some APIs are not authorized, please  [Subscribe](https://developer.tuya.com/en/docs/iot/applying-for-api-group-permissions?id=Ka6vf012u6q76#title-2-Subscribe%20to%20cloud%20products) then [Authorize](https://developer.tuya.com/en/docs/iot/applying-for-api-group-permissions?id=Ka6vf012u6q76#title-3-Authorize%20projects%20to%20call%20the%20cloud%20product). The following APIs must be subscribed for this tutorial: <ul><li>Authorization</li><li>Smart Home Devices Management</li><li>Smart Home Family Management</li><li>Smart Home Scene Linkage</li><li>Smart Home Data Service</li><li>Device status notification</li></ul>|
 
-These services are available for the `tuya` component:
+## Related Documents
 
-- force_update
-- pull_devices
+### How to Contribute
+* [How to Develop a New Driver](https://github.com/tuya/tuya-home-assistant/wiki/How-to-Develop-a-New-Driver)
+* [How to Get the Log](https://github.com/tuya/tuya-home-assistant/wiki/How-to-Get-the-Log)
+* [Supported Device Categories](https://github.com/tuya/tuya-home-assistant/wiki/Supported-Device-Category)
 
-Devices state data and new devices will refresh automatically. If you want to refresh all devices information or get new devices related to your account manually, you can call the `force_update` or `pull_devices` service.
+### Documentation
+* [How does the Integration work](https://github.com/tuya/tuya-home-assistant/wiki/How-the-Integration-works)
+* [Set up Home Assistant on a Raspberry Pi](https://github.com/tuya/tuya-home-assistant/wiki/Set-up-Home-Assistant-on-a-Raspberry-Pi)
+* [FAQs](https://github.com/tuya/tuya-home-assistant/wiki/FAQs)
```

Comment:
Shouldn't the FAQ be part of the documentation here?

```suggestion
- [FAQs](https://github.com/tuya/tuya-home-assistant/wiki/FAQs)
```

Reply:
fixed.